core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -148,38 +148,21 @@ private[v1] trait BaseAppResource extends ApiRequestContext {
 }
 
 private[v1] class ForbiddenException(msg: String) extends WebApplicationException(
-  Response.status(Response.Status.FORBIDDEN).entity(msg).build())
+    UIUtils.buildErrorResponse(Response.Status.FORBIDDEN, msg))
 
 private[v1] class NotFoundException(msg: String) extends WebApplicationException(
-  new NoSuchElementException(msg),
-    Response
-      .status(Response.Status.NOT_FOUND)
-      .entity(ErrorWrapper(msg))
-      .build()
-)
+    new NoSuchElementException(msg),
+    UIUtils.buildErrorResponse(Response.Status.NOT_FOUND, msg))
 
 private[v1] class ServiceUnavailable(msg: String) extends WebApplicationException(
-  new ServiceUnavailableException(msg),
-  Response
-    .status(Response.Status.SERVICE_UNAVAILABLE)
-    .entity(ErrorWrapper(msg))
-    .build()
-)
+    new ServiceUnavailableException(msg),
```

Comment:
Hmm... none of this is your fault, but it seems these exceptions are not needed anymore. jax-rs 2.1 has exceptions for all these and they could just replace these wrappers.

e.g. if you just throw `ServiceUnavailableException` directly instead of this wrapper, the result would be the same.

You can find all exceptions at:
https://jax-rs.github.io/apidocs/2.1/javax/ws/rs/package-summary.html

I think it's better to clean this up instead of adding another helper API.

Reply:
thanks for your review. Sure I can do the refactor, but the helper method would be needed anyway. Indeed, the problem which was present before the PR is that we are not specifying the type of the response, so it takes the type which is produced. And in the default exceptions you mentioned the `Response` is built without setting the type, so we still need to pass them the `Response` we build in the helper method.

I will follow your suggestion (so I'll clean up our exceptions using the jax ones), but keeping the helper method for this reason. Thanks.

Reply:
Hmm, if that's the case then probably the wrappers can save some code (since they abstract the building of the response)...

Reply:
yes, indeed they do. Not too much code but
```
throw new ForbiddenException(raw"""user "$user" is not authorized""")
```
would become
```
throw new ForbiddenException(UIUtils.buildErrorResponse(
            Response.Status.FORBIDDEN, raw"""user "$user" is not authorized"""))
```

what do you think? Shall we do the refactor?

Reply:
I think it's ok to leave (almost) as is. It avoid duplicating the status code in every call, which would be annoying.

But we can remove the "cause" exceptions from all these wrappers since they're redundant.

Reply:
I agree, I updated with your suggestion, thanks